### PR TITLE
[DOC serializer] convert api examples to native classes/Octane

### PR DESCRIPTION
**Pull request by maxwondercorn**
Updated serializer examples - reference #7256.

### CODE REVIEW

File: packages/serializer/addon/-private/transforms/transform.js [hunk 1]
```diff
@@ -45,31 +45,32 @@ import EmberObject from '@ember/object';
...
+  export default class PostModel extends Model {
+    @attr('string') title;
+    @attr('markdown', {
       markdown: {
         gfm: false,
         sanitize: true
       }
     })
+    markdown;
   });
```
Comment (snewcomer) [inline packages/serializer/addon/-private/transforms/transform.js:57]:
```suggestion
  }
```

File: packages/serializer/addon/json.js [hunk 2]
```diff
@@ -32,9 +32,9 @@ import { modelHasAttributeOrRelationshipNamedType } from './-private';
...
 
-  export default Model.extend({
-    friends: hasMany('user'),
-    house: belongsTo('location'),
+  export default class UserModel extends Model {
+    @hasMany('user') friends;
+    @belongsTo('location') house;
 
     name: attr('string')
   });
```
Comment (snewcomer) [inline packages/serializer/addon/json.js:40]:
```suggestion
  }
```

File: packages/serializer/addon/json.js [hunk 3]
```diff
@@ -890,11 +890,11 @@ const JSONSerializer = Serializer.extend({
...
-    export default Model.extend({
-      title: attr(),
-      body: attr(),
+    export default class CommentModel extends Model {
+      @attr title;
+      @attr body;
 
-      author: belongsTo('user')
+      @belongsTo('user') author;
     });
```
Comment (snewcomer) [inline packages/serializer/addon/json.js:898]:
```suggestion
    }
```

**Review by snewcomer — APPROVED** [review]
Super!

File: packages/serializer/addon/-private/transforms/transform.js (hunk 1 above)
Comment (maxwondercorn) [reply packages/serializer/addon/-private/transforms/transform.js:57]: fixed

File: packages/serializer/addon/json.js (hunk 2 above)
Comment (maxwondercorn) [reply packages/serializer/addon/json.js:40]: missed the @attr also - fixed

File: packages/serializer/addon/json.js (hunk 3 above)
Comment (maxwondercorn) [reply packages/serializer/addon/json.js:898]: fixed

**Pull request merged**